docs/proposals/sysctl.md

```diff
@@ -0,0 +1,376 @@
+  `/proc/sys/net/ipv6/conf/all/accept_redirects` (compare
+  [docker#4717](https://github.com/docker/docker/issues/4717#issuecomment-98653017)).
+
+## Constraints and Assumptions
+
+* Only namespaced kernel parameters can be modified
+* Built on-top of the existing security context work
+* Be container-runtime agnostic
+  - on the API level
+  - the implementation (and the set of supported sysctls) will depend on the runtime
+* Kernel parameters can be set during a container creation process only.
+
+## Further work (out of scope for this proposal)
+
+* Update kernel parameters in running containers.
+* Integration with new container runtime proposal: https://github.com/kubernetes/kubernetes/pull/25899.
+* Hugepages support (compare [docker#4717](https://github.com/docker/docker/issues/4717#issuecomment-77426026)) - while also partly configured through sysctls (`vm.nr_hugepages`, compare http://andrigoss.blogspot.de/2008/02/jvm-performance-tuning.html) - is out-of-scope for this proposal as it is not namespaced and as a limited resource (similar to normal memory) needs deeper integration e.g. with the scheduler.
+
+## Use Cases
+
+As an administrator I want to set customizable kernel parameters for a container
+
+1. To be able to limit consumed kernel resources
+   1. so I can provide more resources to other containers
+   1. to restrict system communication that slows down the host or other containers
+   1. to protect against programming errors like resource leaks
+   1. to protect against DDoS attacks.
+1. To be able to increase limits for certain applications while not
+   changing the default for all containers on a host
+   1. to enable resource hungry applications like databases to perform well
+      while the default limits for all other applications can be kept low
+   1. to enable many network connections e.g. for web backends
+   1. to allow special memory management like Java hugepages.
+1. To be able to enable kernel features.
+   1. to enable containerized execution of special purpose applications without
+      the need to enable those kernel features host wide, e.g. ip forwarding for
+      network router daemons
+
+## Community Work
+
+### Docker support for sysctl
+
+Supported sysctls (whitelist) as of Docker 1.12.0:
+
+- IPC namespace
+  - System V: `kernel.msgmax`, `kernel.msgmnb`, `kernel.msgmni`, `kernel.sem`,
+    `kernel.shmall`, `kernel.shmmax`, `kernel.shmmni`, `kernel.shm_rmid_forced`
+  - POSIX queues: `fs.mqueue.*`
+- network namespace: `net.*`
+
+Error behaviour:
+
+- not whitelisted sysctls are rejected:
+
+```shell
+$ docker run --sysctl=foo=bla -it busybox /bin/sh
+invalid value "foo=bla" for flag --sysctl: sysctl 'foo=bla' is not whitelisted
+See 'docker run --help'.
+```
+
+Applied changes:
+
+* https://github.com/docker/docker#19265
+* https://github.com/docker/engine-api#38
+
+Issues:
+
+* https://github.com/docker/docker#21126
```

Comment:
Bad link, can't jump to the issue. better use: ".../issue/19265", like others in the doc.